transports/src/udp_transport.rs

```diff
@@ -0,0 +1,95 @@
+use crate::udp_chunking::SimpleChunker;
+use crate::Transport;
+use anyhow::{anyhow, bail, Result};
+use messages::Message;
+use std::net::{ToSocketAddrs, UdpSocket};
+use std::sync::{Arc, Mutex};
+use std::thread::sleep;
+use std::time::Duration;
+use tracing::{debug, error};
+
+pub struct UdpTransport {
+    pub socket: UdpSocket,
+    mtu: u16,
+    chunker: Arc<Mutex<SimpleChunker>>,
```

Comment:
Curious about the necessity for this mutex?

Reply:
I was looking for an interior mutability solution and this was where I landed. The `UdpTransport` is implementing `Transport`, which requires `Send + Sync`, because `Transport`s are used in threads. The `Arc<Mutex<T>>` solution seemed like the right one for this job given the send/sync requirements.